spring-vault/src/test/java/com/baeldung/springvault/VaultIntegrationManualTest.java

```diff
@@ -96,23 +82,22 @@ public void givenCredentials_whenSave_thenReturnCredentials() {
     }
 
     @Test
-    @Ignore
     public void givenId_whenFindById_thenReturnCredentials() {
         // Given
-        Credentials credentials = new Credentials("login", "p@ssw@rd");
-        Mockito.when(credentialsRepository.findById("login"))
-            .thenReturn(Optional.of(credentials));
+        Assume.assumeTrue("v1".equals(API_VERSION));
+        Credentials expectedCredentials = new Credentials("login", "p@ssw@rd");
+        credentialsService.saveCredentials(expectedCredentials);
 
         // When
-        Optional<Credentials> returnedCredentials = credentialsService.findById("login");
+        Optional<Credentials> retrievedCredentials = credentialsService.findById(expectedCredentials.getUsername());
 
         // Then
-        assertNotNull(returnedCredentials);
-        assertNotNull(returnedCredentials.get());
-        assertEquals(returnedCredentials.get()
-            .getUsername(), credentials.getUsername());
-        assertEquals(returnedCredentials.get()
-            .getPassword(), credentials.getPassword());
+        assertNotNull(retrievedCredentials);
+        assertNotNull(retrievedCredentials.get());
+        assertEquals(retrievedCredentials.get()
+            .getUsername(), expectedCredentials.getUsername());
```

Comment:
Swap order of arguments.

Reply:
Thanks, @KevinGilmore, for catching this. I've fixed the issue.